common/src/main/java/tachyon/security/login/AppLoginModule.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security.login;
+
+import java.util.Map;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+import tachyon.Constants;
+import tachyon.security.User;
+
+/**
+ * An app login module that creates a user based on the user name provided through application
+ * configuration. Specifically, through Java system property tachyon.security.username.
+ * This module is useful if multiple Tachyon clients running under same OS user name
+ * want to get different identifies (for resource and data management), or if Tachyon clients
+ * running under different OS user names want to get same identify.
+ */
+public final class AppLoginModule implements LoginModule {
+  private Subject mSubject;
+  private User mUser;
+
+  @Override
+  public void initialize(Subject subject, CallbackHandler callbackHandler,
+      Map<String, ?> sharedState, Map<String, ?> options) {
+    mSubject = subject;
+  }
+
+  /**
+   * Retrieve the user name by querying the property of Constants.TACHYON_SECURITY_USERNAME.
+   *
+   * @return true if user name provided by application is set and not empty.
+   * @throws javax.security.auth.login.LoginException
+   */
+  @Override
+  public boolean login() throws LoginException {
+    //TODO: after TachyonConf is refactored into Singleton, we will use TachyonConf
+    //instead of System.getProperty for retrieving user name.
+    String userName = System.getProperty(Constants.TACHYON_SECURITY_USERNAME, "");
```

Comment:
hi @calvinjia , I am sorry that I can't understand your mean. which one should be changed to no default?